File: server/internal/dto/request/address.go
package request

type CreateAddressRequest struct {
	Name      string `json:"name" binding:"required,max=32"`
	Phone     string `json:"phone" binding:"required,len=11"`
	Province  string `json:"province" binding:"required,max=32"`
	City      string `json:"city" binding:"required,max=32"`
	District  string `json:"district" binding:"required,max=32"`
	Detail    string `json:"detail" binding:"required,max=256"`
	IsDefault bool   `json:"is_default"`
}

type UpdateAddressRequest struct {
	Name      string `json:"name" binding:"omitempty,max=32"`
	Phone     string `json:"phone" binding:"omitempty,len=11"`
	Province  string `json:"province" binding:"omitempty,max=32"`
	City      string `json:"city" binding:"omitempty,max=32"`
	District  string `json:"district" binding:"omitempty,max=32"`
	Detail    string `json:"detail" binding:"omitempty,max=256"`
	IsDefault *bool  `json:"is_default"`
}

File: client/gusui-user/src/services/address.ts
import { get, post, put, del } from './request'

export interface Address {
  id: number
  name: string
  phone: string
  province: string
  city: string
  district: string
  detail: string
  is_default: boolean
}

export const getAddressList = () =>
  get<Address[]>('/addresses')

export const getAddressDetail = (id: number) =>
  get<Address>(`/addresses/${id}`)

export const createAddress = (data: Omit<Address, 'id'>) =>
  post<Address>('/addresses', data)

export const updateAddress = (id: number, data: Partial<Address>) =>
  put<Address>(`/addresses/${id}`, data)

export const deleteAddress = (id: number) =>
  del(`/addresses/${id}`)

export const setDefaultAddress = (id: number) =>
  put<void>(`/addresses/${id}/default`)
